src/cli.c

```diff
@@ -0,0 +1,53 @@
+#include "cli.h"
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <getopt.h>
+#include <stdio.h>
+
+#define MAX_HZ (1000000L)
+
+static const char* usage_str = "";
+
+__attribute__((__const__)) config_t cli_config_default(void) {
```

Comment:
Maybe this function could return a pointer instead of an instance that is copied over when returning. This could mitigate some heavy copying once the config_t gets bigger.
That would require, e.g., to store a static pointer in cli.c and give functions such as `cli_config_default` (initialization) and `cli_config_destroy` (cleanup)

Reply:
I thought about that, too, but deemed it not necessary right now, as it really is very small as of now. 
However, that could be done pretty easily once necessary.

Reply:
That's done, if you want to have a look at that.